functions/src/modules/SE.functions.ts

```diff
@@ -186,3 +186,92 @@ export const createTaskFromChunks = functions.database.ref(
   });
   return snapshot.val();
 });
+
+
+
+/**
+ * Sends a notification email to the coordinator & udpates the corresponding Task
+ */
+const seUploadsBucketUrl = functions.config().sound_editing.uploads.bucket_name;
+export const processUploadedFile = functions.storage.bucket(seUploadsBucketUrl).object()
+.onFinalize(async object => {
+
+  const filePath = object.name;
+  const seUploadsBucket = admin.storage().bucket(seUploadsBucketUrl);
+  let uploadCode, list, taskId, task, taskRef;
+
+  if (!filePath.startsWith('restored'))
+    return -1;
+
+  try {
+    const filePathRegex = /restored\/(\w+)\/(?:\w+\/)*(([\w]+)-\d+-\d+)\.flac/;
+    const match = filePathRegex.exec(filePath);
+
+    // [Check #1] File name should match `$taskId.flac` pattern.
+    if (!match) 
+      throw new Error(`Wrong Path -- ${filePath} will be deleted.`);
+
+    uploadCode = match[1];
+    taskId = match[2];
+    list = match[3];
+
+    const user = await db.ref(`/users`)
+      .orderByChild('uploadCode')
+      .equalTo(uploadCode)
+      .once('value');
+    
+    if (!user.exists()) 
+      throw new Error(`No User with the given upload code -- ${filePath} will be deleted.`);
+    
+
+    taskRef = await db.ref(`/sound-editing/tasks/${list}/${taskId}`).once('value');
+    task = taskRef.val();
+    
+    // [Check #2] The task should be found in the database by Id.
+    if (!taskRef.exists()) 
+      throw new Error(`Task does not exist -- ${filePath} will be deleted.`);
+
+    
+    // [Check #3] The task should be assigned to a particular sound engineer 
+    //  which is identified by the `uploadCode`within the file path.
+    if (task.restoration.assignee.emailAddress !== user.val().emailAddress)
+      throw new Error(`Wrong upload code -- ${filePath} will be deleted.`);
+
+    // [Check #4] The task should be in `Spare` or `Revise` status.
+    if (['Revise', 'Spare'].indexOf(task.restoration.status) < 0) 
+      throw new Error(`Incorrect task status (only [Revise OR Spare] are allowed here) -- ${filePath} will be deleted.`);
+    
+
+    // Move current file (after passing all validity checks) into `Restored` bucket
+    seUploadsBucket.file(object.name).move(
+      admin.storage().bucket(functions.config().sound_editing.restoration.bucket_name) //add to README
```

Comment:
Construct restoration bucket name here according to https://trello.com/c/59mJXDBs/73-storage-structure.